include/lsst/afw/image/detail/StorableMap.h

```diff
@@ -168,6 +168,38 @@ class StorableMap final {
      */
     bool contains(std::string const& key) const;
 
+    /**
+     * Return `true` if this map contains a mapping for the specified key.
+     *
+     * This is equivalent to testing whether `at(key)` would succeed.
+     *
+     * @tparam T the type of element being tested for
+     * @param key the key to search for
+     *
+     * @return `true` if this map contains a mapping from the specified key to
+     *         a `T`
+     *
+     * @exceptsafe Provides strong exception safety.
+     *
+     */
+    template <typename T, typename std::enable_if_t<std::is_base_of<Storable, T>::value, int> = 0>
+    bool contains(Key<std::shared_ptr<T>> const& key) const {
+        static_assert(std::is_const<T>::value,
+                      "Due to implementation constraints, pointers to non-const are not supported.");
+        if (_contents.count(key.getId()) > 0) {
+            mapped_type const& pointer = _contents.at(key.getId());
+            if (pointer == nullptr) {  // Can't confirm type with dynamic_cast
```

Comment:
Yeah, this is a bit of weirdness in the spec for `StorableMap` (and `GenericMap` before it), caused by my inability to store type information inside the map.

Because a null pointer can be converted (implicitly, no less!) to a pointer to any type, it's impossible to say that a null pointer is not of a particular type without compile-time type information. Thus, the following code is bizarre, but will work:

    map.insert(makeKey<shared_ptr<SkyWcs const>>("extraWcs"s), nullptr);
    if (map.contains(makeKey<shared_ptr<Psf const>>("extraWcs"s))) {
        // The map has a PSF!
        // Who cares if it's null...?
    }

Reply:
I changed the wording to "Null pointer stored; dynamic_cast will always return null."

Reply:
Much better wording, thank you!